### PR TITLE
feat: add tooltip to map markers, add reset button to map, add legend to map

### PR DESCRIPTION
**Pull request by Zerdados**
Takes care of ODDSK-85

Preview here: https://deploy-preview-32--open-data-dashboard.netlify.app/resource/3

### CODE REVIEW

File: src/components/visualization/map/GeoJSONMap.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        geoJsonData && <GeoJSON data={geoJsonData} pointToLayer={pointToLayer} onEachFeature={onEach} />
+      }
+      <ResetView resetViewInput={{ title: 'Reset', pos: standardPos }} />
+      <Legend legendInput={{ title: t('legendTitle'), labels: collectedLabels }} />
+    </MapContainer>
+  );
+}
+
+// TODO: We need to somehow set the information which field of the properties holds the label for the legend.
```
Comment (danielptv) [inline src/components/visualization/map/GeoJSONMap.tsx:39]: Yes, I agree. This solution can only be temporary. As you suggested, the best way if probably adding a configuration option for this.
Regarding colors we probably could also let the user configure it. Either way random colors make for an inconsistent user experience.
Both points can be addressed in a follow-up user story.

File: src/components/visualization/map/mapstyles.module.css
Comment (danielptv) [inline src/components/visualization/map/mapstyles.module.css:1]: As far as I can see, most of these could be replaced by Bootstrap classes. Probably, we do not need a separate CSS file then.

File: src/types/visualization.d.ts
Comment (danielptv) [inline src/types/visualization.d.ts:1]: In my opinion, there is no need to explicitly declare interfaces that are used for component props. Maybe we should declare them inline similar to the `Table` and `ChartTableFilter` components, for example? This is a matter of taste, though.

File: src/components/visualization/map/ResetView.tsx
Comment (danielptv) [inline src/components/visualization/map/ResetView.tsx:1]: The reset icon is misaligned a little bit vertically. Can be improved in a follow up ticket.

File: src/components/visualization/map/Legend.tsx
Comment (danielptv) [inline src/components/visualization/map/Legend.tsx:1]: The circle is not center-aligned with the label (i.e. "Kinos") horizontally. Can be fixed in a follow-up ticket.

File: src/components/visualization/layout/Visualization.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export default function Visualization({ resource }: { resource: Resource }) {
     return <ChartTableWrapper resource={resource} transformedData={transformedData} />;
   } else if (resource.type === 'GeoJSON') {
     const geoJsonData = data as GeoJSON.FeatureCollection;
-    const Map = dynamic(() => import('@/components/visualization/Map'), {
+    const Map = dynamic(() => import('@/components/visualization/map/GeoJSONMap'), {
```
Comment (danielptv) [inline src/components/visualization/layout/Visualization.tsx:25]: If the component is now called `GeoJSONMap` the `const` should have the same name. (Although I did prefer the simpler name `Map`, personally)

**Review by danielptv — COMMENTED** [review]
Great job overall, especially considering how hard extending the Leaflet layout seems to be.

I have left some comments, mostly regarding the styling and the points you had already marked as `TODOs`. Imo, its fine to address those in a follow-up user story.

File: src/components/visualization/layout/Visualization.tsx (hunk 2 above)
Comment (Zerdados) [reply src/components/visualization/layout/Visualization.tsx:25]: Ah, good point. Yes, I also preferred Map, this unfortunately led to problems with using a regular Map inside of GeoMap for collecting the labels.

File: src/components/visualization/map/mapstyles.module.css
Comment (Zerdados) [reply src/components/visualization/map/mapstyles.module.css:1]: Would you be okay with adressing this in the same ticket as the alignment of the icons and texts?

Comment (danielptv) [reply src/components/visualization/map/mapstyles.module.css:1]: Yes, sounds good

File: src/types/visualization.d.ts
Comment (Zerdados) [reply src/types/visualization.d.ts:1]: Good point

**Review by danielptv — APPROVED** [review]
Looks good. No further remarks.

**Pull request merged**